From 6722332168a750bfd295cd44906854669a3fcc68 Mon Sep 17 00:00:00 2001
From: bas smit <bas@baslab.org>
Date: Thu, 25 Jun 2020 16:53:22 +0200
Subject: [PATCH] build: EL7 support

---
 cmake/FindLibBcc.cmake | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/cmake/FindLibBcc.cmake b/cmake/FindLibBcc.cmake
index 9d30b04b..6448e3e1 100644
--- a/cmake/FindLibBcc.cmake
+++ b/cmake/FindLibBcc.cmake
@@ -35,6 +35,7 @@ find_library (LIBBCC_LIBRARIES
     bcc
   PATHS
     /usr/lib
+    /usr/lib64
     /usr/local/lib
     /opt/local/lib
     /sw/lib
@@ -46,6 +47,7 @@ find_library (LIBBCC_BPF_LIBRARY_STATIC
     bcc_bpf
   PATHS
     /usr/lib
+    /usr/lib64
     /usr/local/lib
     /opt/local/lib
     /sw/lib
@@ -57,6 +59,7 @@ find_library (LIBBCC_LOADER_LIBRARY_STATIC
     bcc-loader-static
   PATHS
     /usr/lib
+    /usr/lib64
     /usr/local/lib
     /opt/local/lib
     /sw/lib
-- 
2.25.0

