From a14932640b6d317c6c228822d0399778409ded0c Mon Sep 17 00:00:00 2001
From: bas smit <bas@baslab.org>
Date: Mon, 2 Mar 2020 16:18:08 +0100
Subject: [PATCH] Add lib iberty dependency for static builds

---
 src/CMakeLists.txt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index be18122..7d0076b 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -74,6 +74,9 @@ if(STATIC_LINKING)
   target_link_libraries(bpftrace ${LIBBCC_LIBRARIES})
   target_link_libraries(bpftrace ${LIBBPF_LIBRARY_STATIC})
   target_link_libraries(bpftrace ${LIBBCC_LOADER_LIBRARY_STATIC})
+  if(HAVE_BFD_DISASM)
+    target_link_libraries(bpftrace -liberty)
+  endif(HAVE_BFD_DISASM)
 
   add_library(LIBELF STATIC IMPORTED)
   set_property(TARGET LIBELF PROPERTY IMPORTED_LOCATION ${LIBELF_LIBRARIES})
-- 
2.25.0

