From f760115e6a32cc81e342cf06fb753ac4f106ccd2 Mon Sep 17 00:00:00 2001
From: bas smit <bas@baslab.org>
Date: Fri, 21 Aug 2020 17:23:57 +0200
Subject: [PATCH] Don't require libbpf for build

---
 cmake/FindLibBcc.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/FindLibBcc.cmake b/cmake/FindLibBcc.cmake
index d1be5551..d1d5a1fa 100644
--- a/cmake/FindLibBcc.cmake
+++ b/cmake/FindLibBcc.cmake
@@ -79,7 +79,7 @@ if(STATIC_LINKING)
   find_package(LibBpf)
   find_package(LibElf)
   find_package(LibZ)
-  SET(CMAKE_REQUIRED_LIBRARIES ${LIBBCC_BPF_LIBRARY_STATIC} ${LIBBPF_LIBRARIES} ${LIBELF_LIBRARIES} ${LIBZ_LIBRARIES})
+  SET(CMAKE_REQUIRED_LIBRARIES ${LIBBCC_BPF_LIBRARY_STATIC} ${LIBELF_LIBRARIES} ${LIBZ_LIBRARIES})
 else()
   SET(CMAKE_REQUIRED_LIBRARIES ${LIBBCC_LIBRARIES})
 endif()
--
2.27.0
