From e84e551975e4922ef498c6cc369a18c60becc6fc Mon Sep 17 00:00:00 2001
From: bas smit <bas@baslab.org>
Date: Thu, 7 Nov 2019 20:30:06 +0100
Subject: [PATCH] Add lib iberty dependency for static builds

---
 src/CMakeLists.txt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 65f3a34..ddb55b3 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -62,6 +62,9 @@ if (STATIC_LINKING)
   target_link_libraries(bpftrace ${LIBBCC_LIBRARIES})
   target_link_libraries(bpftrace ${LIBBPF_LIBRARY_STATIC})
   target_link_libraries(bpftrace ${LIBBCC_LOADER_LIBRARY_STATIC})
+  if(HAVE_BFD_DISASM)
+    target_link_libraries(bpftrace -liberty)
+  endif(HAVE_BFD_DISASM)
 else()
   target_link_libraries(bpftrace ${LIBBCC_LIBRARIES})
 endif()
-- 
2.23.0

